### PR TITLE
chore(deps): update gopsutil to v2.20.9

### PR DESCRIPTION
**Pull request by mikemorris**
Followup to https://github.com/hashicorp/consul/pull/8208 to use an actual release instead of gitsha.

Targeting backport to 1.8.x to resolve `go.mod` conflict in https://github.com/hashicorp/consul/pull/8695

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -78,7 +78,7 @@ require (
 	github.com/prometheus/client_golang v1.4.0
 	github.com/rboyer/safeio v0.2.1
 	github.com/ryanuber/columnize v2.1.0+incompatible
-	github.com/shirou/gopsutil v2.20.6-0.20200630091542-01afd763e6c0+incompatible
+	github.com/shirou/gopsutil v2.20.9+incompatible
```
Comment (mikemorris) [inline go.mod:81]: Is the `+incompatible` flag a concern here? What does it mean?

Comment (rboyer) [reply go.mod:81]: It means that though the module advertises its version is v2, it doesn't file the rest of the Go Modules rules for v2.

**Comment by mikemorris** [comment]
🤞 that bumping `golang.org/x/sys` doesn't bring back the FreeBSD build error previously resolved in https://github.com/hashicorp/consul/pull/7780

**Comment by dnephin** [comment]
It looks like you need to run `make update-vendor` again? check vendor failed

**Review by dnephin — APPROVED** [review]
LGTM

**Pull request merged**